tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,931 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5b0241ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "\n",
+    "from torchrl.data import TensorDict\n",
+    "from torchrl.modules import TensorDictModule, TensorDictSequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "129a6de9-cf97-4565-a229-c05ad18df882",
+   "metadata": {},
+   "source": [
+    "## `TensorDictModule` by examples\n",
+    "Let's learn about `TensorDictModule by exploring some examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d1c188a",
+   "metadata": {},
+   "source": [
+    "### Example 1: Simple usage"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d21a711",
+   "metadata": {},
+   "source": [
+    "Let's imagine we have 2 entries `TensorDict`, a and b and we only want to pass a to our network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "6f33781f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+       "        a_out: Tensor(torch.Size([5, 10]), dtype=torch.float32),\n",
+       "        b: Tensor(torch.Size([5, 4, 3]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\"a\": torch.randn(5, 3), \"b\": torch.zeros(5, 4, 3)}, batch_size=[5]\n",
+    ")\n",
+    "linear = TensorDictModule(nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a_out\"])\n",
+    "linear(tensordict)\n",
+    "assert (tensordict[\"b\"] == torch.zeros(5, 4, 3)).all()\n",
+    "tensordict"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3a818629",
+   "metadata": {},
+   "source": [
+    "We can also do it inplace:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "4d1342d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\"a\": torch.randn(5, 3), \"b\": torch.zeros(5, 4, 3)}, batch_size=[5]\n",
+    ")\n",
+    "\n",
+    "linear = TensorDictModule(nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a\"])\n",
+    "\n",
+    "linear(tensordict)\n",
+    "assert tensordict[\"a\"].shape == torch.Size([5, 10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00035cbd",
+   "metadata": {},
+   "source": [
+    "### Example 2: Multiple inputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06a20c22",
+   "metadata": {},
+   "source": [
+    "Now lets imagine a more complex network that takes 2 entries and average them into a single output:"
```

Comment:
Let's remove the "now"

lets => let's

let's imagine => suppose we have

average => averages